examples/libvirt.yaml

```diff
@@ -19,7 +19,7 @@ libvirt:
     name: tectonic
     ifName: tt0
     ipRange: 192.168.124.0/24
-  imagePath: /path/to/image
+  image: http://aos-ostree.rhev-ci-vms.eng.rdu2.redhat.com/rhcos/images/cloud/latest/rhcos-qemu.qcow2.gz
```

Comment:
Pulling this in Go as part of asset generation seems like it would be reasonably straightforward, and in that case we could cache it (with enough information for [`If-Modified-Since`][1] requests on subsequent runs).  But I'm fine punting optimizations like that to future work.

[1]: https://tools.ietf.org/html/rfc7232#section-3.3

Reply:
Agreed. Let's punt for now though.

Reply:
Testing this today, it looks like (at least on libvirt 3.9.0 with QEMU 2.9.0) you need to unzip the image before you can use it.  But Nginx is currently serving it as an octet-stream:

```console
$ curl -sI http://aos-ostree.rhev-ci-vms.eng.rdu2.redhat.com/rhcos/images/cloud/latest/rhcos-qemu.qcow2.gz | grep -i content-type
Content-Type: application/octet-stream
```

so there's no way to cleanly distinguish between "caller pointed us at a gzipped image" and "caller pointed us at an uncompressed image".  Maybe we can get Nginx updated to use [`gzip_static`][1] so we could request `../latest/rhcos-qemu.qcow2` directly and get gzip as a [`Content-Encoding`][2]?  Or look for the gzip and/or [QCOW magic][3] to decide if we need to unzip?  Or maybe I just have old libraries, and newer ones handle this for me?

[1]: http://nginx.org/en/docs/http/ngx_http_gzip_static_module.html
[2]: https://tools.ietf.org/html/rfc7231#section-3.1.2.2
[3]: https://github.com/openshift/installer/pull/271/files#diff-7b580fb0b40a0e0146425ec6ea883350L23

Reply:
Did this ever work on your system? When I wrote the PR, it worked on mine, but maybe something changed since then. I'll test it again today.

Reply:
> Did this ever work on your system?

Nope.  I should start testing libvirt changes loccally before I approve them...